Core/Object Arts/Dolphin/MVP/Presenters/List/UI.ListBox.cls

```diff
@@ -400,7 +400,7 @@ maybeSelChanging: aSymbol
 	self onSelChanged: curSel!
 
 newMultiSelectionsFromEvent: aMouseEvent
-	"Private - Answer a collection of the <integer> selections that would occur if the `MouseEvent` argument was was passed to the list box control.
+	"Private - Answer a collection of the <integer> selections that would occur if the `MouseEvent` argument was passed to the list box control.
```

Comment:
This method comment is redundant and should just be deleted.

Reply:
It would be good to have a top-level "This is what the code does" not just "this is what each section does".

Reply:
In the specific case of this method, it is has been factored out of another method that has the same comment and has a name that, when considered in context, describes what it is intended to do better than the comment.

More generally, the code should be clear enough that a comment is not needed to explain what it does. Comments can be a code smell that the code is not clear by itself; this is particularly true of sectional comments, which suggest a block of code should be factored out to a separate method with a name that replaces the comment. Comments also have a tendency (since they are technically redundant information that is not executed and testable) to be wrong and not to keep pace with change in the code. 

The general principles now intended to be applied are:
1) Public APIs must have a comment describing the what something does (not how).
2) Private APIs should generally not have a comment. If they need one, the method name is probably not clear enough.**
3) In some cases some explanation of non-obvious implementation details may be needed, but only if the intent is not easy to express in the code and would otherwise be lost.
4) Any description of the "how" should form part of a separate comment to the initial "what", but always bearing in mind (3).

** For auto-generated code different rules would apply since the maintenance issues are different, and it isn't generally possible to know at generation time whether there is value in the documentation accompanying the artefact from which the code is being generated.

Different principles have been applied over the years. The code has been added by a variety of authors with different views, and those views have changed over time. I really don't think there is sufficient value in changing all that in isolation, rather it makes sense to clean up when working in the area. In fact I would prefer not to change it unless the code is being worked on, as the style of commenting is sometimes useful historical context. Correcting all the typos, grammatical errors, and obviously wrong/irrelevant ones is fine of course.